### DISCUSSION TITLE
About pseudo-header in iphdr

### DISCUSSION THREAD

**Discussion by systil3** [ANLAB-KAIST/KENSv3#165] [Project Q&A 2. Transport Layer (TCP)] [answered]
In rfc tcp protocol doc(https://datatracker.ietf.org/doc/html/rfc9293), the ip header should include these informations to make pseudo-header, so that we use it to compute checksum for tcp segment;

                +--------+--------+--------+--------+
                |           Source Address          |
                +--------+--------+--------+--------+
                |         Destination Address       |
                +--------+--------+--------+--------+
                |  zero  |  PTCL  |    TCP Length   |
                +--------+--------+--------+--------+
                
but I couldn't find the zero, PTCL, TCP length in iphdr structure. How can I use this?

**Comment by Pusnow** [comment, marked as answer]
For pseudo-header (or checksum), you can use `tcp_sum ` method.

https://github.com/ANLAB-KAIST/KENSv3/wiki/Tip:-Networking-Utilities#tcp-checksum-calculation

http://anlab-kaist.github.io/KENSv3/doc/class_e_1_1_network_util.html#ab83e3dfeed40da8ca13f3702bf91184a
